lib/ansible/cli/pull.py

```diff
@@ -123,8 +125,13 @@ def parse(self):
                                help='modified files in the working repository will be discarded')
         self.parser.add_option('--track-subs', dest='tracksubs', default=False, action='store_true',
                                help='submodules will track the latest changes. This is equivalent to specifying the --remote flag to git submodule update')
+        # add the check_opts group manually, as the full set shortcodes conflict with above --checkout/-C
         self.parser.add_option("--check", default=False, dest='check', action='store_true',
                                help="don't make any changes; instead, try to predict some of the changes that may occur")
+        self.parser.add_option('--syntax-check', dest='syntax', action='store_true',
```

Comment:
im not sure syntax check makes any sense in this context, why do it via pull? the playbook should be checked by ansible-playbook, using pull as a proxy for a check seems contrived.

Reply:
It is. This change works around the Check CLI Option Group being blocked by the `-C` flag being already present in the ansible-pull specifig flags. I was originally going for the `--diff` option, but ended up including all flags from the check group. I can remove it of course.

Reply:
I removed the syntax checking option from this PR and used the opportunity to squash my commits and rebase to latest devel.